File: model/src/main/kotlin/net/dungeonhub/model/discord_role_group/DiscordRoleGroupModel.java
package net.dungeonhub.model.discord_role_group;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import me.taubsie.dungeonhub.common.entity.model.Model;
import me.taubsie.dungeonhub.common.model.discord_role.DiscordRoleModel;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class DiscordRoleGroupModel implements Model {
    private long id;
    private DiscordRoleModel discordRole;
    private DiscordRoleModel roleGroup;
}

File: model/src/main/kotlin/net/dungeonhub/model/discord_role_group/DiscordRoleGroupModel.kt
package net.dungeonhub.model.discord_role_group

import net.dungeonhub.structure.model.Model
import net.dungeonhub.model.discord_role.DiscordRoleModel

class DiscordRoleGroupModel(
    val id: Long,
    val discordRole: DiscordRoleModel,
    val roleGroup: DiscordRoleModel
) : Model